Changed mentions of 'symfony2' to 'symfony'

a-ast opened the pull request:
| Q | A |
| --- | --- |
| Branch? | 3.0+ |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | n/a |
| License | MIT |
| Doc PR | n/a |

Mentions of Symfony2 could be confusing for developers.
I changed 
1. Surrogate capability name from 'symfony2' to 'symfony'
2. User Agent of BrowerKit Client from 'Symfony2 BrowserKit' to 'Symfony BrowserKit'
3. 'symfony2 is fun' in StopwatchExtensionTest. Not only Symfony2 is fun. Symfony3 is even more fun.

a-ast committed:
Changed mentions of 'symfony2' to 'symfony'

- Changed surrogate capability name to symfony
- Changed user agent of the BrowserKit Client to 'Symfony BrowserKit'
- Changed 'symfony2' to 'symfony' in timing templates
- Updated changelog and upgrade files

fabpot commented on src/Symfony/Component/HttpKernel/HttpCache/Esi.php:
```diff
@@ -84,7 +84,7 @@ public function hasSurrogateCapability(Request $request)
     {
         $current = $request->headers->get('Surrogate-Capability');
-        $new = 'symfony2="ESI/1.0"';
+        $new = 'symfony3="ESI/1.0"';
```
If we were to rename this, it would be just `symfony`.

a-ast replied:
I agree it is better. I've changed my PR.

hhamon replied:
Aren't that some sort of a BC break?

a-ast replied:
@hhamon 
Well, the specification says that the capability name [identifies a device token](https://www.w3.org/TR/edge-arch/), not an application or a server. 
'Device tokens must be unique within a request's Surrogate-Capabilities header'.  So, it is more likely an identifier inside a header.

Symfony processed this header in that way:
```php
return false !== strpos($value, 'ESI/1.0');
```
other application should probably process 'Surrogate-Capability' header in the similar way.

Do you have ideas where this change could be a BC?

hhamon replied:
I don't exactly know, I was just wondering. But if an application is strictly comparing the header value, then it will require a change in the application code or in the reverse proxy configuration I suppose. But as this is from the built-in HttpCache, it's not supposed to be used in production, so the impact of breaking something should be very small I guess.

fabpot replied:
Nobody should rely on the name, but @a-ast can you add an entry in the CHANGELOG/UPGRADE files to document that change?

iltar commented:
What about the other occurrences? There's a total of 6 references to symfony2 in the master atm. https://github.com/symfony/symfony/search?utf8=%E2%9C%93&q=symfony2 Might as well take those along while at it ^^

chalasr commented:
As pointed out by @iltar, the tests corresponding to the classes changed here must be updated, the `AbstractSurrogate` class too.

The other occurrences being `src/Symfony/Component/BrowserKit/Client.php` and two tests (`src/Symfony/Bridge/Twig/Tests/Extension/StopwatchExtensionTest.php` and `src/Symfony/Component/BrowserKit/Tests/ClientTest.php`) I think they could be fixed here btw.

a-ast commented:
@iltar, @chalasr, thank you for your comments.
Originally I debugged ESI, found this surrogate capability with 'symfony2' and decided to update it.
However, I agree that it make sense to eliminate all 'symfony2' mentions.

I've done that and changed both PR name and description.
Could you have a look please?

chalasr reviewed (approved):
👍  Looks good to me!

hhamon commented on UPGRADE-3.2.md:
```diff
@@ -71,6 +71,19 @@ HttpKernel
  * `DataCollector::varToString()` is deprecated and will be removed in Symfony
    4.0. Use the `cloneVar()` method instead.
 
+ * Surrogate name in a `Surrogate-Capability` HTTP request header has been changed to 'symfony'.
```
... has been changed to `symfony`.

fabpot commented:
@a-ast Can you rebase this PR and squash your commits? Thanks.

dunglas commented on UPGRADE-3.2.md:
````diff
@@ -71,6 +71,19 @@ HttpKernel
+
+   After:
+   ```
+   Surrogate-Capability: symfony="ESI/1.0"
````
Isn't this a BC break?

dunglas commented on src/Symfony/Component/BrowserKit/Client.php:
```diff
@@ -123,7 +123,7 @@ public function insulate($insulated = true)
     {
         $this->server = array_merge(array(
-            'HTTP_USER_AGENT' => 'Symfony2 BrowserKit',
+            'HTTP_USER_AGENT' => 'Symfony BrowserKit',
```
It's a BC break too.

dunglas commented on src/Symfony/Component/BrowserKit/Tests/ClientTest.php:
```diff
@@ -432,7 +432,7 @@ public function testFollowRedirectWithHeaders()
         $headers = array(
             'HTTP_HOST' => 'www.example.com',
-            'HTTP_USER_AGENT' => 'Symfony2 BrowserKit',
+            'HTTP_USER_AGENT' => 'Symfony BrowserKit',
```
Isn't this a BC break too? It's very common for servers to identify the client using this header.

a-ast commented:
I've updated CHANGELOG/UPGRADE files with information about the BrowserKit user agent change.
Rebased, squashed.
Should I mark my changes in CHANGELOGs with [BC BREAK] labels?

fabpot commented:
LGTM as is

dunglas commented:
👍

fabpot commented:
Thank you @a-ast.
